src/services/account-states/account-state-engine.ts

```diff
@@ -1,118 +1,189 @@
+      newStateObject,
+      event,
+      AccountStateEngine.configuration.edges[transition]?.interventionName,
+    );
+  }
 
-    let stateChange;
-    let interventionName: string | undefined;
+  /**
+   * Helper method to find the name of the state given a state object.
+   * It returns the name if found, throws an error otherwise
+   * @param state - StateDetail object representing the state of the account
+   */
+  private findAccountStateName(state: StateDetails) {
+    for (const key of Object.keys(transitionConfiguration.nodes))
+      if (compareAccountStates(transitionConfiguration.nodes[key]!, state)) return key;
+    throw buildError(
+      MetricNames.STATE_NOT_FOUND_IN_CURRENT_CONFIG,
+      'Account state does not exists in current configuration.',
+    );
+  }
 
-    if (
-      proposedTransition === EventsEnum.AUTH_PASSWORD_RESET_SUCCESSFUL ||
-      proposedTransition === EventsEnum.IPV_IDENTITY_ISSUED
-    ) {
-      const userActionEventObject = AccountStateEngine.userLedActionConfiguration[proposedTransition];
-      if (!userActionEventObject) {
-        logAndPublishMetric(MetricNames.USER_ACTION_EVENT_DOES_NOT_EXIST_IN_CURRENT_CONFIG);
-        throw new StateTransitionError(`User action ${proposedTransition} event does not exist in current config.`);
-      }
-      if (!userActionEventObject.allowedFromStates.includes(accountStateEventEnum)) {
-        logAndPublishMetric(MetricNames.STATE_TRANSITION_NOT_ALLOWED_OR_IGNORED);
-        throw new StateTransitionError(
-          `User action ${proposedTransition} is not allowed on the current account state ${accountStateEventEnum}.`,
-        );
-      }
-      const newState = Object.assign({ ...currentAccountStateDetails }, userActionEventObject.state);
-      newState.suspended = newState.resetPassword || newState.reproveIdentity;
-      stateChange = newState;
-    } else {
-      const currentState = AccountStateEngine.interventionConfigurations[accountStateEventEnum];
-      if (!currentState) {
-        logAndPublishMetric(MetricNames.INTERVENTION_EVENT_NOT_FOUND_IN_CURRENT_CONFIG);
-        throw new StateTransitionError(`State enum ${accountStateEventEnum} does not exist in current config.`);
-      }
-      if (!currentState.allowedTransitions.includes(proposedTransition)) {
-        logAndPublishMetric(MetricNames.STATE_TRANSITION_NOT_ALLOWED_OR_IGNORED);
-        throw new StateTransitionError(
-          `Intervention ${proposedTransition} is now allowed on the current account state ${accountStateEventEnum}.`,
-        );
-      }
+  /**
+   * Helper method to return the list of possible transitions given a state name
+   * If none can be found it throws an error
+   * @param nodeKye - the state name used as the key to identify the nodes in the config object
+   */
+  private findPossibleTransitions(nodeKye: string) {
+    const allowedTransition = AccountStateEngine.configuration.adjacency[nodeKye];
+    if (!allowedTransition)
```

Comment:
`allowedTransition` is an array, right? I think the variable name should be plural. And I'm pretty sure `![]` will return false, so you'll probably need to update the if to `allowedTransitions.length` (or something to that effect).

Reply:
thanks - will change the name to plural. But the condition is right, I am checking for when the variable is undefined overall, not array is empty. Empty array is valid, as a state may have no transitions allowed from it :)